lib/ansible/modules/cloud/azure/azure_rm_networkinterface.py

```diff
@@ -172,7 +172,8 @@
         default: False
     create_with_security_group:
         description:
-            - Specifies whether a default security group should be be created with the NIC. Only applies when creating a new NIC.
+            - Specifies whether a security group should be be created with the NIC.
+            - If this flag set to C(True) and no C(security_group) set, a default security group will be created.
```

Comment:
The issue is complaining  about security_group was ignored. how about adding `if this flag true, security_group will be ignored, a default security group will be created` etc.

Reply:
The logic is 
```python
if self.create_with_security_group:
     security_group = security_group or default_security_group
     create_nsg_instance(security_group)
```